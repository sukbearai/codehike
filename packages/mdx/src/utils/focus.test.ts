import { describe, it, expect, test } from "vitest"

import {
  mapFocusToLineNumbers,
  relativeToAbsolute,
} from "./focus"

test("map focus to line numbers", () => {
  const focus = "1:2,3[1:5,7]"
  const result = mapFocusToLineNumbers(focus, [1, 2, 3, 4])
  expect(result).toMatchSnapshot()
})

describe("relative to absolute", () => {
  it("empty string", () => {
    expect(relativeToAbsolute(undefined, 3)).toBe("3")
    expect(relativeToAbsolute("", 3)).toBe("3")
  })
  it("columns", () => {
    expect(relativeToAbsolute("[1]", 3)).toBe("3[1]")
    expect(relativeToAbsolute("[2,5:7]", 3)).toBe(
      "3[2,5:7]"
    )
  })
  it("lines", () => {
    expect(relativeToAbsolute("(1)", 3)).toBe("3")
    expect(relativeToAbsolute("(2)", 3)).toBe("4")
    expect(relativeToAbsolute("(1,3)", 3)).toBe("3,5")
    expect(relativeToAbsolute("(1:3)", 3)).toBe("3:5")
    expect(relativeToAbsolute("(1:3,4)", 3)).toBe("3:5,6")
    expect(relativeToAbsolute("(1:2,4[1:3,5])", 3)).toBe(
      "3:4,6[1:3,5]"
    )
  })
})
