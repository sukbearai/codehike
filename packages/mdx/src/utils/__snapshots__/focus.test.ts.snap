// Vitest Snapshot v1

exports[`map focus to line numbers 1`] = `
{
  "1": true,
  "2": true,
  "3": [
    {
      "end": 5,
      "start": 1,
    },
    {
      "end": 7,
      "start": 7,
    },
  ],
}
`;
